src/Parser/Warlock/Affliction/Modules/SoulShards/SoulShardTracker.js

```diff
@@ -3,13 +3,20 @@ import Combatants from 'Parser/Core/Modules/Combatants';
 
 import SPELLS from 'common/SPELLS';
 
+const MAX_SHARDS = 5;
+
 class SoulShardTracker extends Analyzer {
   static dependencies = {
     combatants: Combatants,
   };
 
+  _current = 3;   // you regenerate 3 shards out of combat so you should be starting combat always with 3 shards
```

Comment:
If it's possible to manipulate shards prepull somehow this value could be thrown off for the whole encounter. You could get the amount of shards from the energize and cast events like you do with mana in LowMana.js instead of manually counting them to ensure the value is correct.

Reply:
Unfortunately no. Cast events don't contain Soul Shards as a resources (`classResources` only contain mana for us) and energize events don't show current amount of shards (they only show `resourceChange` as in how many shards were gained, and `waste`). 

I agree with you that the starting shards can be manipulated prepull (summoning pet for example) but I don't know if there's any reliable way to track how many did you have on fight start. But usually even when you wipe, you summon your pet, use your Rune and then spend 10 seconds getting food buff. So unless you arrive at the boss, summon pet and *immediately* pull the boss, you should have enough time to regenerate the 3rd shard.

Reply:
That's weird that agony soulshard gains show mana instead of souldshards. But the UA cast events does look like they have the info, so you should be able to atleast check on UA cast to make sure the _current number is correct. 

![image](https://user-images.githubusercontent.com/6773230/35476949-b1270a08-03b9-11e8-89f9-f7e134ec23ff.png)

Reply:
Fixed, I currently assume the fight started with 3 shards but correctly set that info on nearest shard-spending-spell-cast (which should have the class resources related to Soul Shards). If something goes wrong, I subtract 1 shard as usual.